### PR TITLE
HOTFIX for param_export

### PR DESCRIPTION
**Pull request by bkueng**
Seems like #8391 has been merged a bit too quickly.

This fixes a corner case where param saving failed because the buffer was too small by 1 byte. In that case all params are lost.

In addition it reduces the buffer size from 512 to 256 bytes, which will avoid buffer overflows when `param_save_default` is called from other places (for example temperature calibration).

Here are some timing results (all times in microseconds):
```
Buffer size,        export times (multiple runs)
512:                26373, 30756, 26505, 28248
256:                31545, 34755, 37166, 36336
512 + usleep:       213256, 208465, 214441
no buffer + usleep: 767825, 779391
```

It also shows the contribution of removing the `usleep` vs. adding the buffer.

### CODE REVIEW

File: src/modules/systemlib/param/param.c [hunk 1]
```diff
@@ -1140,14 +1139,14 @@ param_export(int fd, bool only_unsaved)
...
-		result = bson_encoder_fini(&encoder); // this will call fsync
+		result = bson_encoder_fini(&encoder);
 
 		// write and finish
 		if ((result != 0) || write(fd, encoder.buf, encoder.bufpos) != encoder.bufpos) {
 			PX4_ERR("param write error");
 
 		} else {
-			px4_fsync(fd);
+			fsync(fd);
```
Comment (dagar) [inline src/modules/systemlib/param/param.c]: I'm not sure why this was done. https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/bson/tinybson.c#L59

**Comment by dagar** [comment]
How did you come across this? I'm close to having the CI system run testing under address sanitizer, but there are a number of issues to work through.

File: src/modules/systemlib/param/param.c (hunk 1 above)
Comment (bkueng) [reply src/modules/systemlib/param/param.c]: This goes back to 9f3bf8e9f4b8f5cde42786211280628168246f99

**Comment by bkueng** [comment]
> How did you come across this? I'm close to having the CI system run testing under address sanitizer, but there are a number of issues to work through.

I reviewed the code and it did not look right. Then I ran some tests with various buffer sizes and found that it indeed failed sometimes. Note that address sanitizer would not have found this because there's no invalid memory access. The bson parser would just abort, since the buffer was too small and `realloc_ok` is false.

**Pull request merged**